src/UI/MainMenuOptions.cpp

```diff
@@ -0,0 +1,370 @@
+{
+    text(constants::WINDOW_SYSTEM_TITLE);
+    init();
+}
+
+
+MainMenuOptions::~MainMenuOptions()
+{
+    auto& e = NAS2D::Utility<NAS2D::EventHandler>::get();
+    e.keyDown().disconnect(this, &MainMenuOptions::onKeyDown);
+    btnOk.click().disconnect(this, &MainMenuOptions::btnOkClicked);
+    btnCancel.click().disconnect(this, &MainMenuOptions::btnCancelClicked);
+    btnApply.click().disconnect(this, &MainMenuOptions::btnApplyClicked);
+}
+
+
+void MainMenuOptions::init()
+{
+
+    auto& e = NAS2D::Utility<NAS2D::EventHandler>::get();
+    e.keyDown().connect(this, &MainMenuOptions::onKeyDown);
+
+    size(500, 350);
+
+    const auto border_left_width = 5;
+    const auto border_right_width = 5;
+    const auto border_top_height = 25; //WINDOW_TITLE_HEIGHT + 5
+    const auto border_bottom_height = 5;
+    const auto element_spacing = 5;
+    const auto left_edge = positionX() + border_left_width;
+    const auto right_edge = left_edge + width() - border_right_width * 2.0f;
+    const auto top_edge = positionY() + border_top_height;
+    const auto bottom_edge = top_edge + height() - border_top_height - border_bottom_height * 2.0f;
+    const auto width = right_edge - left_edge;
+    const auto height = bottom_edge - top_edge;
+
+    pnlControls.position(left_edge, top_edge);
+    pnlControls.size(width, height);
+
+    pnlButtons.position(pnlControls.positionX(), bottom_edge - element_spacing);
+    pnlButtons.size(pnlControls.width(), 120);
+
+    lblResolution.text("Resolution");
+    lblResolution.size(0, 0);
+
+    cmbResolution.visible(true);
+    cmbResolution.enabled(true);
+    cmbResolution.size(200, 0);
+    cmbResolution.maxDisplayItems(1);
+
+    cmbResolution.clearSelection();
+    //TODO: add functions to the renderer/DeviceContext to get/set display count and modes.
+    {
+        extern SDL_Window* underlyingWindow;
+        const auto display_index = SDL_GetWindowDisplayIndex(underlyingWindow);
+        const auto num_resolutions = SDL_GetNumDisplayModes(display_index);
+        std::ostringstream ss;
+        std::string resolutionText{};
+        SDL_DisplayMode active_mode{};
+        for(int i = 0; i < num_resolutions; ++i)
+        {
+            SDL_DisplayMode cur_mode{};
+            SDL_GetDisplayMode(display_index, i, &cur_mode);
+            SDL_GetCurrentDisplayMode(display_index, &active_mode);
+            ss << cur_mode.w << 'x' << cur_mode.h << 'x' << cur_mode.refresh_rate;
+            if(cur_mode.w == active_mode.w && cur_mode.h == active_mode.h && cur_mode.refresh_rate == active_mode.refresh_rate)
+            {
+                currentResolutionSelection = i;
+                resolutionText = ss.str();
+            }
+            cmbResolution.addItem(ss.str(), i);
+            ss.str("");
+        }
+        //TODO: Add void currentSelection(int) to ComboBox
+        //  OR
+        //TODO: Add ComboBox::text override to set internal text field value.
```

Comment:
I don't much like see `TODO` comments left around. I assume this is for selecting the current video mode as the default/current selection.